examples/Google.Cloud.Functions.Examples.IntegrationTests/GoogleChatBotTest.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Functions.Testing;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Google.Cloud.Functions.Examples.IntegrationTests;
+
+public class GoogleChatBotTest
+{
+    [Fact]
+    public async Task CanChat()
+    {
+        using (var server = new FunctionTestServer<GoogleChatBot.TypedFunction>())
```

Comment:
You shouldn't need to do that. Make the class derive from `FunctionTestBase<GoogleChatBot.TypedFunction>` and you can use `ExecuteHttpRequestAsync` directly:

```csharp
    JToken expected = ... // See below
        string requestContent = @"{
            ""message"": {
                ""sender"": {
                    ""displayName"": ""janedoe"",
                    ""imageUrl"": ""example.com/avatar.png""
                }
            }
        }";
        var request = new HttpRequestMessage
        {
            Method = HttpMethod.Post,
            Content = new StringContent(requestContent, Encoding.UTF8, "application/json")
        };
        await ExecuteHttpRequestAsync(request, async response =>
        {
            JToken actual = JToken.Parse(await response.Content.ReadAsStringAsync());
            Assert.True(JToken.DeepEquals(actual, expected), $"JSON response {actual} did not match expected {expected}");
        });
```